boot/zephyr/include/mcuboot_config/mcuboot_config.h

```diff
@@ -154,9 +154,38 @@
 
 #endif /* !__BOOTSIM__ */
 
+#if CONFIG_NRFX_WDT && CONFIG_BOOT_WATCHDOG_FEED
```

Comment:
This seems imbalanced.  There really are two conditions where we have to default: enabled without a vendor implementation, and disabled.  I think the high-level structure should be:
```
#if CONFIG_BOOT_WATCHDOG_FEED
#if VENDOR
...
#elif OTHER_VENDOR
...
#else /* VENDORS */
/* No vendor implementation, no-op for historical reasons but should be a configuration error */
#define MCUBOOT_WATCHDOG_FEED() do { } while (0)
#endif /* VENDORS */
#else /* CONFIG_BOOT_WATCHDOG_FEED */
 /* Not enabled, no feed activity */
#define MCUBOOT_WATCHDOG_FEED() do { } while (0)
#endif /* CONFIG_BOOT_WATCHDOG_FEED */
```
because as proposed the entry condition doesn't pair correctly with the top-level else, and the preprocessor conditional nesting is more complex than it needs to be.